src/Security/BasicAuth.php

```diff
@@ -82,10 +85,12 @@ public static function requireLogin($realm, $permissionCode = null, $tryUsingSes
 
         $member = null;
         if (isset($_SERVER['PHP_AUTH_USER']) && isset($_SERVER['PHP_AUTH_PW'])) {
-            $member = MemberAuthenticator::authenticate(array(
+            $authenticator = Injector::inst()->get(Authenticator::class);
+
+            $member = $authenticator->authenticate([
                 'Email' => $_SERVER['PHP_AUTH_USER'],
```

Comment:
>Make sense, although we'll probably want to clarify the keys that authenticate() expects

I'm not sure that's really required... the authenticator can just do a lookup with DataList::filter... which is pretty much [how Laravel does it](https://github.com/laravel/framework/blob/e1fc1fa860566c5f9346376fbf896e7954749f48/src/Illuminate/Auth/EloquentUserProvider.php#L93-L111). Laravel has a concept of a auth provider (the driver to fetch users) which is in charge of resolving the user from credentials and the persistence layer (the DB).

If our convention is to have a unique ID field that's configurable, we should try to make use of it.